### PR TITLE
Hw wk02 d01

### PR DESCRIPTION
**Pull request by mkikegawa**
{
  "comfortability": 3,
  "completeness": 3.5,
  "comment: I was chasing my tail over the passing of methods - reverted to previous states multiple times. Ironically the code I had previously submitted worked the best. I tried to make this more complicated than it needed to be. 
WIP:
- Shopping List Class and integration with robot
- Complete the Author & User class integration
  -  Complete Robot Logic

I would prefer to do a class check for the author but i have not implemented that yet. 
}

### CODE REVIEW

**Comment by mkikegawa** [comment]
{
"comfortability": 1,
"completeness": 1,
"comment:  completed basic talkingrobot.rb the files to review are in the redo folder. 

I am completely lost on the rspec lab. The most basic checks on the Person class are testing ok but I am just following the existing pattern. 
}

File: lib/Person.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
+require 'pry'
```
Comment (rgpass) [inline lib/Person.rb:1]: File name should be all snake_case

File: original/Author.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+require 'pry'
+require './Person'
+
+class Author < Person
+  def initialize(args = {})
+    super
```
Comment (rgpass) [inline original/Author.rb]: Clearer to write `super(args)`

**Comment by rgpass** [comment]
There are a lot of files here. Let's talk at the break to get this figured out.

**Comment by mkikegawa** [comment]
@rgpass  I have updated directory structure

File: lib/Author.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require 'pry'
```
Comment (rgpass) [inline lib/Author.rb:1]: Go with snake_case for all file names.

File: lib/Person.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    puts "What's your name?"
+    gets.chomp.capitalize
+  end
+
+  def self.get_user_age
+    puts"How old are you?"
+    gets.chomp.to_i
+  end
+
+  def self.get_user_gender
```
Comment (rgpass) [inline lib/Person.rb:14]: A sign this belongs somewhere else: you use a different class name in the variable name
`get_user_name` --> put this in `User` and call it `get_name`

File: lib/User.rb [hunk 5]
```diff
@@ -0,0 +1,211 @@
+require 'pry'
+require './Person'
+require './GroceryList'
+
+class User < Person 
+  attr_accessor :name, :age, :gender
+  def initialize(args = {})
+    super(args)
+      @spaghetti     = args[:spaghetti]
+    # @answers   = args[:answers]
```
Comment (rgpass) [inline lib/User.rb:10]: Better to pass in a GroceryList instance when you initialize it. That way you get access to all the methods inside the `GroceryList` class!

File: lib/User.rb [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+      puts "You're 75!!!!"
+    when 76..150
+      puts "You turned 75 #{delta} years ago."
+    end
+  end
+
+  def get_grandparent_status?      
+    has_grandkids = gets.chomp.upcase
+    has_grandkids == "Y"
+  end
```
Comment (rgpass) [inline lib/User.rb:36]: Is this specific to users? I think it applies to all people. Put this guy in the Person class along with all the other methods that aren't specific to users.

File: lib/talkative_robot.rb [hunk 7]
```diff
@@ -0,0 +1,284 @@
+require "./User"
+require "./Author"
+require "./GroceryList"
+
+# require 'pry'
```
Comment (rgpass) [inline lib/talkative_robot.rb:5]: Remove unused comments.

File: spec/user_spec.rb [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (rgpass) [inline spec/user_spec.rb:28]: So much space.

**Comment by rgpass** [comment]
:+1: Can add more tests and rearrange some stuff, but overall it looks good.

When done, run these dudes: `$ git checkout master` `$ git pull` `$ git merge <branchnameForThisHomework>` `$ git push`

**Pull request merged**